Mathlib/Data/Matroid/Minor/Basic.lean

```diff
@@ -534,6 +534,187 @@ lemma Dep.of_contract (h : (M ／ C).Dep X) (hC : C ⊆ M.E := by aesop_mat) : M
 
+/-! ### Finiteness -/
+
+instance contract_finite [M.Finite] : (M ／ C).Finite := by
+  rw [← dual_delete_dual]
+  infer_instance
+
+instance contract_rankFinite [RankFinite M] : RankFinite (M ／ C) :=
+  let ⟨B, hB⟩ := (M ／ C).exists_isBase
+  ⟨B, hB, hB.indep.of_contract.finite⟩
+
+instance contract_finitary [Finitary M] : Finitary (M ／ C) := by
+  obtain ⟨J, hJ⟩ := M.exists_isBasis' C
+  suffices h : (M ／ J).Finitary by
+    rw [hJ.contract_eq_contract_delete]
+    infer_instance
+  exact ⟨fun I hI ↦ hJ.indep.contract_indep_iff.2  ⟨disjoint_left.2 fun e heI ↦
+    ((hI {e} (by simpa) (by simp)).subset_ground rfl).2,
+    indep_of_forall_finite_subset_indep _ fun K hK hKfin ↦
+      (hJ.indep.contract_indep_iff.1 <| hI (K ∩ I)
+      inter_subset_right (hKfin.inter_of_left _)).2.subset (by tauto_set)⟩⟩
+
+/-! ### Loops and Coloops -/
+
+lemma contract_eq_delete_of_subset_loops (hX : X ⊆ M.loops) : M ／ X = M ＼ X := by
+  simp [(empty_isBasis_iff.2 hX).contract_eq_contract_delete]
+
+lemma contract_eq_delete_of_subset_coloops (hX : X ⊆ M.coloops) : M ／ X = M ＼ X := by
+  rw [← dual_inj, dual_delete, contract_eq_delete_of_subset_loops hX, dual_contract]
+
+@[simp]
+lemma contract_isLoop_iff_mem_closure : (M ／ C).IsLoop e ↔ e ∈ M.closure C ∧ e ∉ C := by
+  obtain ⟨I, hI⟩ := M.exists_isBasis' C
+  rw [hI.contract_eq_contract_delete, delete_isLoop_iff, ← singleton_dep,
+    hI.indep.contract_dep_iff, singleton_union, hI.indep.insert_dep_iff, hI.closure_eq_closure]
+  by_cases heI : e ∈ I
+  · simp [heI, hI.subset heI]
+  simp [heI, and_comm]
+
+@[simp]
+lemma contract_loops_eq (M : Matroid α) (C : Set α) : (M ／ C).loops = M.closure C \ C := by
+  simp [Set.ext_iff, ← isLoop_iff, contract_isLoop_iff_mem_closure]
+
+@[simp]
+lemma contract_coloops_eq (M : Matroid α) (C : Set α) : (M ／ C).coloops = M.coloops \ C := by
+  rw [← dual_delete_dual, dual_coloops, delete_loops_eq, dual_loops]
+
+@[simp]
+lemma contract_isColoop_iff : (M ／ C).IsColoop e ↔ M.IsColoop e ∧ e ∉ C := by
+  simp [isColoop_iff_mem_coloops]
+
+lemma IsNonloop.of_contract (h : (M ／ C).IsNonloop e) : M.IsNonloop e := by
+  rw [← indep_singleton] at h ⊢
+  exact h.of_contract
+
+@[simp]
+lemma contract_isNonloop_iff : (M ／ C).IsNonloop e ↔ e ∈ M.E \ M.closure C := by
+  rw [isNonloop_iff_mem_compl_loops, contract_ground, contract_loops_eq]
+  refine ⟨fun ⟨he,heC⟩ ↦ ⟨he.1, fun h ↦ heC ⟨h, he.2⟩⟩,
+    fun h ↦ ⟨⟨h.1, fun heC ↦ h.2 ?_⟩, fun h' ↦ h.2 h'.1⟩⟩
+  rw [← closure_inter_ground]
+  exact (M.subset_closure (C ∩ M.E)) ⟨heC, h.1⟩
+
+lemma IsBasis.diff_subset_loops_contract (hIX : M.IsBasis I X) : X \ I ⊆ (M ／ I).loops := by
+  rw [diff_subset_iff, contract_loops_eq, union_diff_self,
+    union_eq_self_of_subset_left (M.subset_closure I)]
+  exact hIX.subset_closure
+
+/-! ### Closure -/
+
+/-- Contracting the closure of a set is the same as contracting the set,
+and then deleting the rest of its elements. -/
+lemma contract_closure_eq_contract_delete (M : Matroid α) (C : Set α) :
```

Comment:
Just checking it's intentional that M and C are explicit in this lemma and the next, since they're not in most of the lemmas after this one

Reply:
Yes it is - in other cases they can be inferred.